script/version_up.py

```diff
@@ -0,0 +1,666 @@
+#!/usr/bin/env python3
+# Copyright (c) 2024 The Brave Authors. All rights reserved.
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this file,
+# You can obtain one at https://mozilla.org/MPL/2.0/.
+"""Version upgrade command line tool.
+
+This tool is used to upgrade Brave to a desired version of Chromium. The goal
+is to produce a set of patches for the new chromium base version, that looks
+something like:
+
+1. Update from Chromium [from] to [to].
+2. Conflict-resolved patches from Chromium [from] to [to].
+3. Update patches from Chromium [from] to [to].
+4. Updated strings for Chromium [to].
+
+The process is started by providing a base branch in Brave to be used as the
+starting point for the base Chromium version, and a target version to be used.
+
+- script/version_up.py --previous=origin/master --to=135.0.7037.1
+
+The workflow with this script:
+
+1. The first commit will be created where package.json is updated, and so is
+   the pinslist timestamp source.
+2. The repository will be initialised with the newer version of Chromium,
+   followed where all patches will be applied on the newer version.
+3. Any patches that fail to apply will be reattempted with the --3way option.
+4. If any patches still fail to apply, the process will stop, and the user will
+   be asked to resolve the conflicts manually for the files that cannot be
+   resolved. In such a case, the run will stop.
+5. If any patches are deleted, the process will be stopped and the user will be
+    asked to commit the deleted patches manually, preserving the history of why
+    they are deleted.
+5. Having resolved all conflicts, the tool will be run again with the same
+   arguments, and with the added --continue flag.
+6. This tool will then continue from the point where it stopped, staging all
+   patches that were applied, and committing them as conflict-resolved patches.
+7. This tool will then continue to update the patches, and rebase the strings
+   for the new version of Chromium.
+
+Steps 3-7 may end up being skipped altogether if no failures take place, or in
+part if resolution is possible without manual intervention.
+
+Additionally, this tool can be run with the --update-patches-only flag, which
+will can be used to generated the "Update patches" and "Updated strings"
+commits on their own. This useful when rebasing branches, and regenerating
+these patches is desired.
+"""
+
+import argparse
+from datetime import datetime
+import json
+from pathlib import Path
+import re
+import subprocess
+import sys
+
+# This file is updated whenever the version number is updated in package.json
+PINSLIST_TIMESTAMP_FILE = 'chromium_src/net/tools/transport_security_state_generator/input_file_parsers.cc'
+VERSION_UPGRADE_FILE = ".version_upgrade"
```

Comment:
Not a Python expert, but should this string use single quotes as the string above it does?

Reply:
Fixed.